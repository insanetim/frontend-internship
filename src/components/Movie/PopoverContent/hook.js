import { useDispatch, useSelector } from 'react-redux'

import { showModal } from 'src/state/app/actions'
import { addToList } from 'src/state/lists/actions'
import { listsSelector } from 'src/state/lists/selectors'

export const useContainer = ({ movieId, setPopoverOpen }) => {
  const dispatch = useDispatch()
  const lists = useSelector(listsSelector)

  const handleAddToNewList = () => {
    dispatch(
      showModal({
        modalType: 'CREATE_LIST_MODAL',
        modalProps: {
          cb: listId => {
            dispatch(addToList({ listId, movieId }))
          }
        }
      })
    )
    setPopoverOpen(false)
  }

  const handleAddToList = listId => {
    dispatch(addToList({ listId, movieId }))
    setPopoverOpen(false)
  }

  return { lists, handleAddToNewList, handleAddToList }
}
