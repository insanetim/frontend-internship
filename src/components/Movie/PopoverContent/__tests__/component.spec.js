import { shallow } from 'enzyme'

import PopoverContent from '../component'

const mockedHookData = {
  lists: {
    results: [{ id: 1, name: 'Test list' }]
  },
  handleAddToNewList: jest.fn(),
  handleAddToList: jest.fn()
}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

it('matches snapshot', () => {
  const component = shallow(<PopoverContent />)

  expect(component).toMatchSnapshot()
})

it('handles button click', () => {
  const component = shallow(<PopoverContent />)
  component.find('Button').last().simulate('click')

  expect(mockedHookData.handleAddToList).toHaveBeenCalledWith(1)
})
