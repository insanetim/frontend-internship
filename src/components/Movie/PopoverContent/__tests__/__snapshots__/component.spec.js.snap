// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <div>
    <Button
      onClick={[MockFunction]}
      type="link"
    >
      Create new list ...
    </Button>
  </div>
  <div
    key="1"
  >
    <Button
      onClick={[Function]}
      type="link"
    >
      Test list
    </Button>
  </div>
</Fragment>
`;
